### DISCUSSION TITLE
Resolver In GraphQL

### DISCUSSION THREAD

**Discussion by EricBui619** [OrchardCMS/OrchardCore#9324] [Q&A] [answered]
I have a custom graphql type based on OC soure code that supports for quering  content item by its content parts. But I got a problem, the Resolver of a field type was triggered more than one when I sent a graphql http request. After checking, I recognize that, the number of response data equal to the one of content items of the content type we query. I guess it will loop for every content item in a particular content type and try to resolve part for creating a graphql object type and then trigger resolver, I'm not sure the exact reason is. Anyone can support me?

![image](https://user-images.githubusercontent.com/30250345/116680507-f47ea200-a9d5-11eb-8f82-582a192b9ee7.png)

**Comment by tropcicstefan** [comment]
That is by design, because every content item needs to be parsed(resolved'). Only way to put less strain on DB is by using DataLoaders(look at `ContentPickerFieldQueryObjectType`)

**Reply by EricBui619** [reply]
@tropcicstefan  I think I've found the reason because I was mistaken. My filterByContentPart is just a field in a content type, even if I add a filter for it, it just get appropriate data in that field. So I have 5 content items, it will loop 5 times, each time it will trigger resolver on the FilterByContentPart filed.

Bwt, do you have any ideas how to add filter by content part in the input filter a content type?

![image](https://user-images.githubusercontent.com/30250345/116773177-192e5480-aa7e-11eb-85a7-7d7b84acc5f3.png)

**Reply by tropcicstefan** [reply]
Don't know how to put it on level above, but for similar situations I used
```
Field<ListGraphType<ContentItemInterface>, IEnumerable<ContentItem>>()
                .Name("contentItems")
                .Description("the content items")
                .Argument<StringGraphType, string>("articleAlias", "alias of original article")
                .ResolveLockedAsync(async g =>
               {
                    var serviceProvider = g.ResolveServiceProvider();
                    var queryService = serviceProvider.GetRequiredService<IQueryService>();

                    var queryParametersDict = new Dictionary<string, object>() {
                        { "articleAlias", g.GetArgument<string>("articleAlias") }
                    };
                    var queryParam = JsonConvert.SerializeObject(queryParametersDict);

                    return await queryService.GetItemsQueryAsync(Query.SummarySections, queryParam);
                });
```

you add argument on part and call it in resolver with GetArgument()
if this is what you meant?

**Reply by EricBui619** [reply]
@tropcicstefan  Simply, I just wanna filter content items by its content part, beside options that are provided by OC. Does your way above support this scenario?

**Comment by EricBui619** [comment, marked as answer]
Hi all, I've implemented feature that allow us to query content items by fields of its content part. Some logic in my code based on code base of OC. The following image is my demo for this feature:

![image](https://user-images.githubusercontent.com/30250345/116838042-20c53900-abf7-11eb-919a-a083b3d40ba8.png)
